docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+major/minor releases. They are normally named ``maint-<version>``.
+
+With patch-level releases, you'll often make multiple releases and work with
+multiple maintenance branches. For instance, you might have to merge a
+bug fix to ``master``, but also to ``maint-3.1`` and to ``maint-3.2``.
+
+Major/minor-level release
+-------------------------
+When releasing a major/minor version, you will have to take extra care to the
+compatibility with the current released ``invenio`` version, by checking
+the upper bounds defined for this module in the ``invenio`` releases.
+
+The most probable scenario is that for a given ``invenio`` release,
+e.g. ``3.3``, the upper bound for an Invenio module ``foo``
+(with current latest release ``1.2.6``) is set to ``invenio-foo<1.3.0``.
+A new minor release ``1.3.0`` of this module will not affect
+``invenio 3.3``.
+
+If this is not the case, you will have to make sure that the new release
+of the module ``foo`` will not introduce backward incompatible changes,
+because it will be installed by ``invenio 3.3``, potentially breaking
+existing Invenio installations.
+
+Module release
+--------------
+
+How to release an Invenio module.
+
+**Python module**
+
+1. Make sure that all relevant PRs have been merged to ``master``.
+2. Make sure that the build on ``master`` branch is passing.
+3. Prepare release notes in
+    - Update the ``CHANGE.rst`` file, making sure that you describe changes,
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``version.py`` (follow semantic versioning).
+5. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+6. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+7. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
+
+**JavaScript/React module**
+
+1. Make sure that all relevant PRs have been merged to ``master``.
+2. Make sure that the build on ``master`` branch is passing.
+3. Prepare release notes in
+    - Update the ``CHANGE.md`` file, making sure that you describe changes,
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``package.json`` (follow semantic versioning).
+5. Regenerate the ``package-lock.json``:
+    - ``rm package-lock.json``
+    - ``npm install``
+6. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+7. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+8. Regularly check ``Travis`` and ``NpmJS`` to ensure the release went through.
+
+**Backport changes to a maint release**
```

Comment:
Comment: I wouldn't call it backport. It can happen in both directions. First fix in maint, then merged to master - or fix in master merge to maint.

Reply:
Perhaps something like "Patch-level releases: moving between branches" or "Releasing bug fixes on multiple branches".

Reply:
I have no strong preference, what is the consensus on this one?

Reply:
I went for `Patch-level releases on multiple maint branches`